tests/libnet/cloudinit.go

```diff
@@ -23,12 +23,106 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+func NewNetworkData(options ...NetworkDataOption) (string, error) {
+	networkData := CloudInitNetworkData{
+		Version: 2,
+	}
+
+	for _, option := range options {
+		err := option(&networkData)
+		if err != nil {
+			return "", nil
+		}
+	}
+
+	nd, err := yaml.Marshal(&networkData)
+	if err != nil {
+		return "", err
+	}
+
+	return string(nd), nil
+}
+
+func WithEthernet(name string, options ...NetworkDataInterfaceOption) NetworkDataOption {
+	return func(networkData *CloudInitNetworkData) error {
+		if networkData.Ethernets == nil {
+			networkData.Ethernets = map[string]CloudInitInterface{}
+		}
+
+		networkDataInterface := CloudInitInterface{name: name}
+
+		for _, option := range options {
+			err := option(&networkDataInterface)
+			if err != nil {
+				return nil
+			}
+		}
+
+		networkData.Ethernets[name] = networkDataInterface
+		return nil
+	}
+}
+
+func WithAddresses(addresses ...string) NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		if networkDataInterface.Addresses == nil {
+			networkDataInterface.Addresses = []string{}
+		}
+		networkDataInterface.Addresses = append(networkDataInterface.Addresses, addresses...)
+		return nil
+	}
+}
+
+func WithDHCP4() NetworkDataInterfaceOption {
```

Comment:
Don't we need an `enabled bool` argument?

Reply:
`WithDHCP4` is clearly activating it if we need deactivation we can add `WithoutDHCP4`

Reply:
Well, that is the default anyway (enabled).
`WithDHCP4` may also mean: With this attribute, so adding a flag makes sense.

But if you prefer it this way, ok.